repository/repository.go

```diff
@@ -23,6 +23,28 @@ var (
 	repoWithNameRegexp = regexp.MustCompile(`^(?P<owner>[A-Za-z0-9_\-]+)/(?P<name>[A-Za-z0-9._\-]+)(?:\((?P<params>.+)\))?$`)
 )
 
+// SearchType represents the type of search to be performed.
+type SearchType int
```

Comment:
why not just use a string, with the user-friendly values? "repositories" and "code"

Reply:
Here i'm using enums instead of just string in order to group my related searchType values together in a clear maintainable and type-safe way, and we need `int` and `iota` to do that.

Then later it allows me to do:
```
func parseSearchType(str string) SearchType {
	if searchType, ok := searchTypeMap[str]; ok {
		return searchType
	}

	return Repositories
}
```

```
		if searchType == Code {
			repos, resp, err = searchCodeRepositories(ctx, ghClient, query, opts, params)
		} else {
			repos, resp, err = searchRepositories(ctx, ghClient, query, opts, params)
		}
```

Reply:
yep, if you want a bit less code you can do something like https://go.dev/play/p/QdFHGd6umbs

Reply:
[This article](https://www.sohamkamani.com/golang/enums/) shows few immediate issues using string instead of iota.
What do you think of it ?